identity-actor/src/actor/endpoint.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::actor::Error;
+use crate::actor::Result as ActorResult;
+use std::borrow::Cow;
+use std::fmt::Display;
+use std::str::Split;
+
+use serde::Deserialize;
+use serde::Serialize;
+
+static HOOK: &str = "hook";
+
+/// A path-like identifier for a handler function. As an example, `identity/create`
+/// could be an endpoint of the "identity" namespace and the concrete action "create".
+///
+/// An endpoint can also represent a `hook` function, indicated by a `/hook` postfix.
+///
+/// Endpoints are separated by slashes and only allow alphabetic ascii characters and `_`.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[serde(try_from = "String", into = "String")]
+pub struct Endpoint {
+  name: Cow<'static, str>,
+}
+
+impl Endpoint {
+  /// Returns whether this endpoint is a hook.
+  pub fn is_hook(&self) -> bool {
+    let split: Split<char> = self.name.as_ref().split('/');
+    split.last().expect("split should have at least one element") == HOOK
```

Comment:
See below, but this method was removed because it references hooks which will most likely no longer be part of the actor itself but implemented on a higher level.